c/xen_common.c

```diff
@@ -885,22 +885,27 @@ static int count_children(xmlNode *n, const char *name)
 static void destring(xen_session *s, xmlChar *name, const abstract_type *type,
                      void *value)
 {
-    switch (type->typename)
```

Comment:
Ok, adding the braces allows it to compile, so the switch block can stay
(In C# it compiles without braces, although they do change the scope of the declared variable).
I've replaced the commited changeset.